### PR TITLE
[React] Hooks (password-reset-finish component)

### PR DESCRIPTION
**Pull request by qmonmert**
WIP on #9891

One component : PasswordResetFinish

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/finish/password-reset-finish.tsx.ejs [hunk 1]
```diff
@@ -16,45 +16,33 @@
...
-    password: '',
-    key: getUrlParameter('key', this.props.location.search)
-  };
-
-  componentWillUnmount() {
-    this.props.reset();
-  }
+  useEffect(() => {
+    return () => props.reset();
+  }, []);
```
Comment (chrisdns) [inline generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/finish/password-reset-finish.tsx.ejs]: could you please change this to `useEffect(() => () => props.reset(), []);
` 
:smile:

File: generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/finish/password-reset-finish.tsx.ejs [hunk 2]
```diff
@@ -16,45 +16,33 @@
...
-export interface IPasswordResetFinishProps extends DispatchProps, RouteComponentProps<{key: string}>  {}
+export interface IPasswordResetFinishProps extends DispatchProps, RouteComponentProps<{key: string}> {}
 
-export interface IPasswordResetFinishState {
-  password: string;
-  key: string;
-}
+export const PasswordResetFinishPage = (props: IPasswordResetFinishProps) => {
+  const [password, setPassword] = useState('');
+  const [key, setKey] = useState(getUrlParameter('key', props.location.search));
```
Comment (chrisdns) [inline generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/finish/password-reset-finish.tsx.ejs]: you can also remove `setKey` which seems to no be used

File: generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/finish/password-reset-finish.tsx.ejs (hunk 1 above)
Comment (qmonmert) [reply generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/finish/password-reset-finish.tsx.ejs]: @chrisdns ok

File: generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/finish/password-reset-finish.tsx.ejs (hunk 2 above)
Comment (qmonmert) [reply generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/finish/password-reset-finish.tsx.ejs]: @chrisdns ok

**Pull request merged**